contents/docker/Docker.md

```diff
@@ -0,0 +1,161 @@
+deployment make Docker an ideal tool when doing testing. You can enable docker
+in your TRAVIS Continous Integration pipeless to do testing for your
+application. 
+
+Docker containers will provide an isolated environment for the
+testing of said application and will always run it on the same configured
+setting. Even if your application gets updated, the Docker container will still
+be able to rollback and test backwards compatibility on an older image.
+
+If you need to perform an upgrade during a product’s release cycle, you can easily make the necessary changes to Docker containers, test them, and implement the same changes to your existing containers. Docker allows you to build, test, and release images that can be deployed across multiple servers. Even if a new security patch is available, the process remains the same. You can apply the patch, test it, and release it to production.
+
+#### 2.5 Docker gives security
+
+From a security point of view, Docker ensures that applications that are running on containers are completely segregated and isolated from each other, granting you complete control over traffic flow and management. No Docker container can look into processes running inside another container. 
+
+This is ideal especially when you dont want Users to exploit vulnerabilities
+between different applications. Docker's isolation ensures that this is not even
+a factor to consider.
+
+### 3.0 When to use docker?
+
+Despite all the abovementioned advantages, it still may not be completely clear
+as to why we need Docker. There are certain situations where it is ideal to make
+use of Docker and 'containerization'.
+
+Some situations you may want to use Docker is as follows:
+
+**Learning new technologies**: To get started with a new tool without spending time on installation and configuration, Docker offers an isolated and disposable environment. Many projects maintain Docker images with their applications already installed and configured. For example, if you want to check out a distributed tracing system like Zipkin, you can have a working setup on localhost simply by running:
+    
+```
+sudo docker run -d -p 9411:9411 openzipkin/zipkin
+```
+
+**Basic use cases**: Pulling images from Docker Hub is also a good solution if your application is basic or standard enough to work with a default Docker image. Cases such as hosting a website using a LAMP stack, using a reverse proxy, or setting up a Minecraft server often have an official or well-supported image available on DockerHub. If the default configuration in these images is acceptable for your needs, then pulling the image can save a lot of time that would otherwise be spent setting up your environment and installing the necessary tools.
+
+**Developer teams**: If you have developers working with different setups, Docker provides a convenient way to have local development environments that closely match the production environment, without needing to ssh into a remote box.
```

Comment:
mention something about `compose`? like how docker allows you to setup multiple things at the same time?